File: JSLib/src/FrameWork/EngineLoop.js
namespace SDK.Lib
{
    /**
     * @brief 主循环
     */
    public class EngineLoop
    {
        public void MainLoop()
        {
            // 处理客户端自己的消息机制
            MsgRouteBase routeMsg = null;
            while ((routeMsg = Ctx.m_instance.m_sysMsgRoute.pop()) != null)
            {
                Ctx.m_instance.m_msgRouteNotify.handleMsg(routeMsg);
            }

            // 处理网络
            if (!Ctx.m_instance.m_netCmdNotify.bStopNetHandle)
            {
                ByteBuffer ret = null;
                while ((ret = Ctx.m_instance.m_netMgr.getMsg()) != null)
                {
                    if (null != Ctx.m_instance.m_netCmdNotify)
                    {
                        Ctx.m_instance.m_netCmdNotify.addOneHandleMsg();
                        Ctx.m_instance.m_netCmdNotify.handleMsg(ret);       // CS 中处理
                        Ctx.m_instance.m_luaSystem.receiveToLuaRpc(ret);    // Lua 中处理
                    }
                }
            }

            // 处理 input
            //Ctx.m_instance.m_inputMgr.handleKeyBoard();
            // 游戏循环处理
            Ctx.m_instance.m_processSys.ProcessNextFrame();
            //Ctx.m_instance.m_logSys.updateLog();
        }
    }
}

File: PyLibs/src/Libs/FrameWork/EngineLoop.py
# -*- coding: utf-8 -*-

'''
@brief 主循环
'''

from Libs.Core.GObject import GObject;
from Libs.FrameWork.Ctx import Ctx;

class EngineLoop(GObject):
    
    def __init__(self):
        super(EngineLoop, self).__init__();
        
        self.mTypeId = "EngineLoop";
        
        
    
    def mainLoop(self):
        # 游戏循环处理
        Ctx.instance().mProcessSys.ProcessNextFrame();
        Ctx.instance().mLogSys.updateLog();

